internal/services/apicenter/apicenter_environment_resource.go

```diff
@@ -0,0 +1,251 @@
+	Identification string `tfschema:"identification"`
+	Description    string `tfschema:"description"`
+	Type           string `tfschema:"environment_type"`
+	DevPortalUri   string `tfschema:"development_portal_uri"`
+	Instructions   string `tfschema:"instructions"`
+	ServerType     string `tfschema:"server_type"`
+	MgmtPortalUri  string `tfschema:"management_portal_uri"`
+}
+
+func (r ApiCenterEnvironmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"service_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"identification": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"environment_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice(
+				environments.PossibleValuesForEnvironmentKind(),
+				false),
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"development_portal_uri": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"instructions": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"server_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice(
+				environments.PossibleValuesForEnvironmentServerType(),
+				false),
+		},
+		"management_portal_uri": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ApiCenterEnvironmentResource) ModelObject() interface{} {
+	return &ApiCenterEnvironmentResourceModel{}
+}
+
+func (r ApiCenterEnvironmentResource) ResourceType() string {
+	return "azurerm_apicenter_environment"
+}
+
+func (r ApiCenterEnvironmentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return environments.ValidateEnvironmentID
+}
+
+func (r ApiCenterEnvironmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			service, err := services.ParseServiceID(model.ServiceId)
+			if err != nil {
+				return fmt.Errorf("parsing ApiCenter Service ID %s: %+v", model.ServiceId, err)
+			}
+
+			// @favoretti: can't find any workspace creation buttons in the portal, assume everything defaults to "default" for now?
+			id := environments.NewEnvironmentID(subscriptionId, service.ResourceGroupName, service.ServiceName, "default", model.Identification)
```

Comment:
The ID is typically constructed using the `name` property, is there a reason we're using `identification` instead?

Reply:
Great question for the Azure API dev team :) It took me a few moments before I realized this was happening... But that's what they are using. I went with an assumption it was name as usual and it gave me a perpetual diff. In the whole flow editable name and identification fields that in the UI default to each other's value until changed makes no sense...